src/main/kotlin/uk/gov/justice/digital/hmpps/visitallocationapi/service/NomisSyncService.kt

```diff
@@ -123,18 +126,49 @@ class NomisSyncService(
 
     if (voBalanceChange != 0 || pvoBalanceChange != 0) {
       LOG.info("Balance has changed as a result of sync for prisoner $prisonerId, for domain event ${domainEventType.value}")
-      dpsPrisoner.changeLogs.add(changeLogService.logSyncEventChange(dpsPrisoner, domainEventType))
+      dpsPrisoner.changeLogs.add(changeLogService.createLogSyncEventChange(dpsPrisoner, domainEventType))
     }
 
     prisonerDetailsService.updatePrisonerDetails(prisoner = dpsPrisoner)
   }
 
-  @Transactional
   fun syncPrisonerRemoved(prisonerId: String) {
     LOG.info("Entered NomisSyncService - syncPrisonerRemoved for prisoner {}", prisonerId)
     prisonerDetailsService.removePrisonerDetails(prisonerId)
   }
 
+  fun getChangeLogForNomis(requestDto: VisitAllocationPrisonerAdjustmentRequestDto): VisitAllocationPrisonerAdjustmentResponseDto {
+    val prisonerChangeLogs = changeLogService.findAllChangeLogsForPrisoner(requestDto.prisonerId).sortedBy { it.id }
+
+    var previousEntry: ChangeLog? = null
+    var currentEntry: ChangeLog? = null
+
+    for (log in prisonerChangeLogs) {
+      if (log.id == requestDto.changeLogId) {
+        currentEntry = log
+        break
+      }
+      previousEntry = log
+    }
```

Comment:
suggestion only - another way of achieving this could be by using indexOfFirst{log.id == requestDto.changeLogId} and previousEntry as index - 1 (if current is > 0). This will help you avoid looping through them all. 
https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/index-of-first.html